lms/static/js/ccx/view/schedule_date_button.js

```diff
@@ -0,0 +1,107 @@
+var define = window.define || RequireJS.define;  // jshint ignore:line
+
+define(
+  [
+    'backbone',
+    'jquery',
+    'underscore',
+    'gettext',
+    'text!templates/ccx/underscore/date-button.underscore'
+  ],
+  function (
+    Backbone,
+    $,
+    _,
+    gettext,
+    scheduleTreeDateButtonTemplate
+  ) {
+    'use strict';
+     return Backbone.View.extend({
+
+       initialize: function(options) {
+        this.dateType = options.dateType;
+        this.date = options.date;
+       },
+
+       render: function () {
+         this.$el.html(_.template(scheduleTreeDateButtonTemplate)({
+           date: this.date
+         }));
+         this.location = this.$el.data("location");
+         var $dateChangeButton = $(this.$el.find("button"));
+
+         $dateChangeButton.attr('href', '#enter-date-modal').
+          leanModal({closeButton: '.close-modal'});
+         $dateChangeButton.click(this.editDateCCXSchedule());
+
+         return this;
+       },
+
+       editDateCCXSchedule: function() {
+         var self = this;
+         return function () {
+           var dateParts;
+           var row = $(this).closest('tr');
+           var modal = $('#enter-date-modal')
+             .data('what', self.dateType)
+             .data('location', row.data('location'));
+
+           modal.find('h2').text(
+             self.dateType === 'due' ? gettext("Enter Due Date and Time") :
+               gettext("Enter Start Date and Time")
+           );
+
+           if (self.date) {
+             dateParts = self.date ? self.date.split(' ') : ['', ''];
+             var date = dateParts[0];
+             var time = dateParts[1];
+
+             modal.find('input[name=date]').val(date);
+             modal.find('input[name=time]').val(time);
+           }
+
+           modal.focus();
+           $(document).on('focusin', function (event) {
+             try {
+               if (!_.isUndefined(event.target.closest('.modal').id) &&
```

Comment:
I do not understand what this code is doing. Can you explain (better yet, add comments)?

And although this appears to be handling moving focus into the modal, there is no visual indication of where focus is when the dialog opens. Also, closing the dialog does not move focus back to a reasonable place (it goes to the top of the page). It should return focus to the element that triggered the dialog.